examples/bri-3/src/bri/transactions/agents/transactions.agent.spec.ts

```diff
@@ -1,330 +1,339 @@
+import { LoggingModule } from '../../../shared/logging/logging.module';
+import { NotFoundException } from '@nestjs/common';
+import { NOT_FOUND_ERR_MESSAGE as WORKSTEP_NOT_FOUND_ERR_MESSAGE } from '../../workgroup/worksteps/api/err.messages';
+import { NOT_FOUND_ERR_MESSAGE as WORKFLOW_NOT_FOUND_ERR_MESSAGE } from '../../workgroup/workflows/api/err.messages';
+import { AuthModule } from '../../../bri/auth/auth.module';
+import { AutomapperModule } from '@automapper/nestjs';
+import { classes } from '@automapper/classes';
+
+let agent: TransactionAgent;
+let authAgent: AuthAgent;
+let prisma = new PrismaClient();
+let bpiSubject1: BpiSubject;
+let bpiSubject2: BpiSubject;
+let bpiSubjectAccount1: BpiSubjectAccount;
+let bpiSubjectAccount2: BpiSubjectAccount;
+let bpiAccount1: BpiAccount;
+let workgroup: Workgroup;
+let workstep: Workstep;
+let workflow: Workflow;
+
+beforeEach(async () => {
+  prisma = new PrismaClient();
+  const module: TestingModule = await Test.createTestingModule({
+    imports: [
+      LoggingModule,
+      AuthModule,
+      AutomapperModule.forRoot({
+        strategyInitializer: classes(),
+      }),
+    ],
+    providers: [
+      TransactionAgent,
+      WorkstepStorageAgent,
+      WorkflowStorageAgent,
+      MerkleTreeService,
+      TransactionStorageAgent,
+      PrismaService,
+      PrismaMapper,
+      BpiSubjectStorageAgent,
+      {
+        provide: 'ICircuitService',
+        useClass: SnarkjsCircuitService,
+      },
+    ],
+  })
+    .overrideProvider(PrismaService)
+    .useValue(prisma)
+    .compile();
+
+  agent = module.get<TransactionAgent>(TransactionAgent);
+  authAgent = module.get<AuthAgent>(AuthAgent);
+  bpiSubject1 = await prisma.bpiSubject.create({
+    data: {
+      name: 'name',
+      description: 'desc',
+      publicKey:
+        '0x047a197a795a747c154dd92b217a048d315ef9ca1bfa9c15bfefe4e02fb338a70af23e7683b565a8dece5104a85ed24a50d791d8c5cb09ee21aabc927c98516539',
+    },
+  });
 
-let transactionAgent: TransactionAgent;
-
-const transactionStorageAgentMock: DeepMockProxy<TransactionStorageAgent> =
-  mockDeep<TransactionStorageAgent>();
-const workstepStorageAgentMock: DeepMockProxy<WorkstepStorageAgent> =
-  mockDeep<WorkstepStorageAgent>();
-const workflowStorageAgentMock: DeepMockProxy<WorkflowStorageAgent> =
-  mockDeep<WorkflowStorageAgent>();
-const authAgentMock: DeepMockProxy<AuthAgent> = mockDeep<AuthAgent>();
-const merkleTreeServiceMock: DeepMockProxy<MerkleTreeService> =
-  mockDeep<MerkleTreeService>();
-const circuitsServiceMock: DeepMockProxy<ICircuitService> =
-  mockDeep<ICircuitService>();
-
-// TODO: #742 Setup of this test data below is what should be handled in a separate file where we mock only prisma.client
-// and implement various test data scenarios that can be selected with a single line of code.
-// https://github.com/demonsters/prisma-mock
-const existingWorkgroupId = uuid();
+  bpiSubject2 = await prisma.bpiSubject.create({
```

Comment:
we have a lot of builiders implemented in builders.ts.
Could we can use them to define the test domain objects we want to work with and then just have separate methods that abstract away prisma.create calls? This way we can share those methods across the test code base.

This way we could probably ditch the as ModelBpiSubjectAccount syntax as well.

Convetion would be that we work with domain objects in tests, and prisma create calls are only there to avoid mocking the storage agents

Reply:
It is mainly for readability and maintainability purposes. I agree it might be too much at this point. Maybe just open an issue and we can introduce them later if time allows.

Reply:
@skosito can you open an issue for this so we can keep track of it? cc @ognjenkurtic @Kasshern

Reply:
@skosito dont forget to open an issue for this one pls.